### PR TITLE
Gas Comparisons

### PR DESCRIPTION
**Pull request by k-xo**
- [x] Add {MAINNET_RPC_URL} as env variable

### CODE REVIEW

**Review by Brean0 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -8,7 +8,11 @@
     "test": "test"
   },
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "t": ". ./script/bin/env.sh; forge t -vvv",
+    "deploy:local": ". ./script/bin/env.sh; ./script/bin/deploy-local.sh",
+    "compare": "forge snapshot --match-path test/integration/IntegrationTestGasComparisons.sol --snap .dex-comparisons",
+    "compare:check": "yarn compare --check",
+    "compare:csv": "node ./script/dex-comparisons.js"
```
Comment (silochad) [inline package.json:15]: @k-xo flagging this change for you, separated the snapshot command from the one that applies `--check` which throws an error in CI if the gas diff changes.

File: test/integration/IntegrationTestGasComparisons.sol [hunk 2]
```diff
@@ -257,6 +263,44 @@ contract IntegrationTestGasComparisons is IntegrationTestHelper {
         vm.warp(block.timestamp + 1);
         daiWethWell.swapFrom(daiWethTokens[1], daiWethTokens[0], 500 * 1e18, 500 * 1e18, address(this));
     }
+
+    enum ClipboardType {
```
Comment (Brean0) [inline test/integration/IntegrationTestGasComparisons.sol:267]: Can we put this into integrationTestHelper?

File: test/integration/IntegrationTestGasComparisons.sol [hunk 3]
```diff
@@ -70,49 +71,54 @@ contract IntegrationTestGasComparisons is IntegrationTestHelper {
...
     function testFuzz_wells_WethDaiUsdc_Swap(uint amountIn) public {
-        
         vm.pauseGasMetering();
         uint amountIn = bound(amountIn, 1e18, 1000e18);
 
-        // approve WETH to send to pipeline
-        WETH.approve(address(pipeline), type(uint).max);
-        
+        WETH.approve(address(depot), type(uint).max);
+        deal(address(WETH), address(depot), amountIn * 5);
```
Comment (Brean0) [inline test/integration/IntegrationTestGasComparisons.sol:78]: remove

**Review by Brean0 — APPROVED** [review]

File: test/integration/IntegrationTestGasComparisons.sol (hunk 2 above)
Comment (k-xo) [reply test/integration/IntegrationTestGasComparisons.sol:267]: Done

File: test/integration/IntegrationTestGasComparisons.sol (hunk 3 above)
Comment (k-xo) [reply test/integration/IntegrationTestGasComparisons.sol:78]: can remove the deal, but we need to keep the approve as the first call of the `farm` call is transferring the tokens from `msg.sender` to `pipeline` via `depot`.

**Review by silochad — APPROVED** [review]

**Pull request merged**